### ISSUE TITLE
Extraneous formatMessage Parameter

### ISSUE THREAD

**Issue by aabenoja** [format-message/babel-plugin-format-message#1]
Consider the following:

```
// en-us.locale.json
{
  "hello": "Hello, {name}!"
}
```

```jsx
export default function MyStatelessComponent(props) {
  return <div>{formatMessage('hello', props)}</div>;
}
```

After babel transpiles my code I get the following:

```jsx
export default function MyStatelessComponent(props) {
  return <div>{$$_hello_someHash(formatMessage, props)}</div>;
}

$$_hello_someHash(formatMessage, args) {
  return 'Hello, ' + args['name'] + '!';
}
```

I understand there are cases where we want to have the actual `formatMessage` function such as when we need date transformations, but I don't think it's necessary to have to import `format-message` into my bundle just to handle some of these interpolations.

**Comment by thetalecrafter** [comment]
`format-message` is not very large in your runtime. Most of its weight is in the cli tools, which won't be included in your bundle.

That being said, I could probably remove the import declaration when it isn't needed, and could probably inline the date/time/number formatting, it will just be a rather large effort.

**Comment by aabenoja** [comment]
You're right, it is pretty small. However, my page failing to load because of a missing argument was very unexpected.

> That being said, I could probably remove the import declaration when it isn't needed...

Wait, the import is supposed to be added by the plugin?

**Comment by thetalecrafter** [comment]
Sorry, I misunderstood. There are two different issues here.

1. The generated function has an unused parameter in some case. Are you linting generated code, and gating the build on the linter? An unused parameter shouldn't cause a page to not load.
2. The generated code relies on `format-message` being imported/required, and therefore included in the web bundle. This is the concern I was addressing in my last comment. For now you must explicitly `require` or `import` format-message in each file you use it. In the future the runtime could possibly be obsoleted and removed, but that would take a lot of work.

**Comment by aabenoja** [comment]
I am not running the linter, only this babel plugin. I'll probably add that to our test pipeline once I check if our unit tests fail on this. The only reason the page doesn't load is because my bundle is for a purely react page. That missing variable error kills the whole bundle.

Thanks for your help. I personally have no issue with importing `format-message`, but I'm sure it would annoy other devs to run into this undocumented caveat.

**Comment by thetalecrafter** [comment]
Can you clarify @aabenoja which problem you are experiencing and what your expected solution would be?
